### ISSUE TITLE
Can you make it so you can compile without being root?

### ISSUE THREAD

**Issue by thpryrchn** [pyther/openwrt-feed#3]
I wasted a whole day trying to get this to compile! 
There are 2 problems that I had to overcome. 
  1. golang of at least 1.13?? needs to be installed on the compiling OS. IT is not listed as a dependency.
  2. It fails if not compiling as root. 

Compiling as root is a bad practice, and every other package in openwrt compiles as user.

I also figured out how to do this through docker. 

Here is what I did to run in docker. https://hub.docker.com/r/openwrtorg/sdk/tags -- select the tag for the target/platform you are compiling for.
```bash
docker run --rm -v [Local folder you want the compiled version to go in.]:/home/build/openwrt/bin -it openwrtorg/sdk:ath79-generic-master

# Become root
sudo -s

# Install golang
wget https://golang.org/dl/go1.15.linux-amd64.tar.gz
sudo tar -C /usr/local -xzf go1.15.linux-amd64.tar.gz
export PATH=$PATH:/usr/local/go/bin

# The rest is from the readme
cp feeds.conf.default feeds.conf
echo 'src-git pyther git://github.com/pyther/openwrt-feed' >> feeds.conf
./scripts/feeds update -a
./scripts/feeds install -a
make menuconfig
make package/goeap_proxy/download && make package/goeap_proxy/prepare
make package/goeap_proxy/compile 
```

**Comment by adagari** [comment]
For what it's worth, when I last compiled this software I did so without having to elevate to root. I was trying to compile in a docker container at first but was running into many issues, so I went ahead and installed the dependence on the host os and it compiled without a problem. When I have some time I will try and reproduce the issue you are having to see if it affects everyone.

**Comment by pyther** [comment]
Hi @thpryrchn. Thanks for reporting the issue. I agree that building as root is never a good idea idea.

It sounds like there may be something wrong with your build environment. I've been building against master every a couple of weeks without any problems. I've never built as root.

The Makefile is very similar to the Makefile of other go applications in OpenWRT. I'm not aware of doing anything special for goeap_proxy Makefile.

**Comment by thpryrchn** [comment]
I tried on a virtual machine running latest Ubuntu, and it was complaning about an option -x with time.. So I tried the docker build sdk for openwrt. I figured that would be the best build enviorment with the littlest configuration. That dock didn't have go installed, so I had to install it, and then it was getting permission failures trying to rm files.. But when doing all the steps as root, it compiles.

openwrt's sdk uses Debian 10.2 as the base OS.

**Comment by pyther** [comment]
@thpryrchn I was able to reproduce the issue in a clean container. I think I know what was wrong, hope to have a fix out shortly.

**Comment by pyther** [comment]
Fixed in acc5996b439561efbfddb5f1c7a2f4c23ce66a18, I was missing `PKG_BUILD_DEPENDS:=golang/host`. 

Added a Dockerfile, so build can be a bit more reproducible. ReadMe has been updated with instructions.

**Comment by thpryrchn** [comment]
Sweet! Looks like I inspired you. This should also help people who aren't that linux familiar..
